src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue

```diff
@@ -52,6 +51,111 @@ export default {
       required: false,
     },
   },
+  computed: {
+    formattedTokens: ({
+      language,
+      formattedSwiftTokens,
+      tokens,
+    }) => (language === Language.swift.key.api ? formattedSwiftTokens : tokens),
+    // Return a formatted version of the tokens array, with additional
+    // indentation whitespace to break parameters onto individual lines for
+    // improved readability and scanning of Swift functions/initializers.
+    //
+    // This is implemented in a single pass loop where text tokens are updated
+    // at certain key points to insert spaces and newlines so that each
+    // parameter of a multi-parameter function gets its own line.
+    //
+    // Ideally this should be implemented with a tool like SwiftFormat in the
+    // future with the raw text before it gets tokenized for syntax
+    // highlighting—however, this post-tokenization JavaScript logic should be
+    // an improvement until that kind of work can be integrated.
+    //
+    // @param {Array} tokens The original syntax tokens.
+    //   See `DeclarationToken.props`
+    // @return {Array} A formatted version of the original tokens.
+    //   See `DeclarationToken.props`
+    formattedSwiftTokens: ({ tokens }) => {
+      const numSpacesForIndent = 4; // maybe this could be configurable in the future
```

Comment:
Maybe something that could be part of theme settings JSON? 👀

Reply:
I think that sounds reasonable. I purposefully wrote this in a way where the number could be easily be updated by something along those lines as opposed to hardcoding in the spaces manually, but maybe I should just go ahead and make that just work.

Reply:
I was initially hesitant to add this until we've been able to make this file more complete and better documented its use cases, but I've added support now with the knowledge that we will soon expose this information in a more discoverable way.

6c9b11d56bd9a4cdb0bda8fcbda399921b0e9aae